crates/bevy_app/src/app.rs

```diff
@@ -765,9 +765,9 @@ impl App {
                 plugin_name: plugin.name().to_string(),
             })?;
         }
-        self.is_building_plugin = true;
+        self.building_plugin_counter += 1;
```

Comment:
It's unlikely, but these should always be checked adds and subtractions. If this overflows (highly unlikely, though still incorrect), it can result in missing the check.

Reply:
I have to disagree, it's not highly unlikely, it's impossible. `usize` is large enough to store every memory address and for each recursion depth, we need to at least keep track of `Box<dyn Plugin>`. We will run out of memory addresses much sooner than we will reach `usize::MAX`.